data/scripts/lib/register_monster_type.lua

```diff
@@ -204,6 +204,11 @@ registerMonsterType.loot = function(mtype, mask)
 			end
 			if loot.subType or loot.charges then
```

Comment:
define charges as
`local charges = loot.charges or ItemType(loot.id):getCharges()`
then just use charges everywhere where loot.charges is used
charges > 0 check is probably redundant as it wont change anything, unless you want to save that single call